ipa-core/src/protocol/ipa_prf/quicksort.rs

```diff
@@ -181,25 +187,34 @@ where
                     let rvl_ctx = rvl_ctx.clone();
                     let record_id = RecordId::from(idx);
                     async move {
-                        // Compare the current element against pivot and reveal the result.
-                        let comparison = compare_gt::<_, ThirtyTwoBitStep, SORT_CHUNK>(
+                        // Compare elements against pivot
+                        let comp = compare_gt::<_, ThirtyTwoBitStep, SORT_CHUNK>(
                             cmp_ctx, record_id, &k, &pivot,
                         )
-                        .await?
-                        .reveal(rvl_ctx, record_id) // reveal outcome of comparison
                         .await?;
-
-                        // desc = true will flip the order of the sort
-                        Ok::<_, Error>(comparison + !desc)
+                        // Reveal the comparison result
+                        reveal(rvl_ctx, record_id, &comp).await
                     }
                 },
                 || (AdditiveShare::<K>::ZERO, AdditiveShare::<K>::ZERO),
             ),
-        )
-        .try_flatten_iters()
-        .map_ok(bool::from)
-        .try_collect()
-        .await?;
+        );
+
+        let unwrapped_compare_results = {
+            // TODO: this should be replaced with a malicious verify.
```

Comment:
@danielmasny ?

Reply:
Same as above, we compute the comparison, validate and then reveal. So we would probably call validated_seq_join on the comparison block or do our own customize seq join in chunks that calls validate in between compare and reveal